Multiplatform/Shared/Sidebar/RefreshProgressModel.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  RefreshProgressModel.swift
+//  NetNewsWire
+//
+//  Created by Phil Viso on 7/2/20.
+//  Copyright © 2020 Ranchero Software. All rights reserved.
+//
+
+import Account
+import Combine
+import Foundation
+import SwiftUI
+
+class RefreshProgressModel: ObservableObject {
+	
+	enum State {
+		case refreshProgress(Float)
+		case lastRefreshDateText(String)
+		case none
+	}
+		
+	@Published var state = State.none
+	
+	private static var dateFormatter: RelativeDateTimeFormatter = {
+		let formatter = RelativeDateTimeFormatter()
+		formatter.dateTimeStyle = .named
+		
+		return formatter
+	}()
+		
+	private let lastRefreshDate: () -> Date?
+	private let combinedRefreshProgress: () -> CombinedRefreshProgress
+	
+	private static let lastRefreshDateTextUpdateInterval = 60
+	private static let lastRefreshDateTextRelativeDateFormattingThreshold = 60.0
+	
+	init(lastRefreshDateProvider: @escaping () -> Date?,
```

Comment:
I tried to keep this model as lightweight as possible instead of directly requiring `AccountManager` to supply its data. I planned to write tests, but it looks like the project doesn't have test targets for the multi-platform stuff yet.